### PR TITLE
Apply default textures from model when passing nil to default.player_set_textures

### LINKED ISSUE #1861: default.player_set_textures only takes effect after player model change
Setting the player textures to nil with `default.player_set_textures` only causes the default textures from the model to be used next time the model is changed using `default.player_set_model`. The documentation implies that the textures will revert to the default textures immediately (which would be expected as setting the textures to non-default textures takes effect immediately).

### PR DESCRIPTION
**Pull request by danielmeek32**
Fixes #1861

### CODE REVIEW

File: mods/default/player.lua [hunk 1]
```diff
@@ -71,8 +71,15 @@ end
 
 function default.player_set_textures(player, textures)
 	local name = player:get_player_name()
+	local model = models[player_model[name]]
+	local model_textures
+	if model then
+		model_textures = model.textures
+	else
+		model_textures = nil
```
Comment (SmallJoker) [inline mods/default/player.lua]: Redundant else case. `model_textures` is already nil if not set.

File: mods/default/player.lua [hunk 2]
```diff
@@ -71,8 +71,15 @@ end
...
+	end
 	player_textures[name] = textures
```
Comment (SmallJoker) [inline mods/default/player.lua]: Why not update this table with the new texture as well? -> `textures or model_textures`

File: mods/default/player.lua (hunk 1 above)
Comment (danielmeek32) [inline mods/default/player.lua]: Fair enough, I am not completely familiar with lua's handling of unset or undefined values. Feel free to remove the `else` section.

File: mods/default/player.lua (hunk 2 above)
Comment (danielmeek32) [inline mods/default/player.lua]: I missed that. Again feel free to add this.

**Comment by danielmeek32** [comment]
I've updated the pull request with the requested changes following the code review.

File: mods/default/player.lua [hunk 3]
```diff
@@ -71,8 +71,13 @@ end
 
 function default.player_set_textures(player, textures)
 	local name = player:get_player_name()
-	player_textures[name] = textures
-	player:set_properties({textures = textures,})
+	local model = models[player_model[name]]
+	local model_textures
+	if model then
+		model_textures = model.textures
+	end
```
Comment (sfan5) [inline mods/default/player.lua]: can be shortened to `local model_textures = model and model.textures or nil`

Comment (danielmeek32) [inline mods/default/player.lua]: Good catch.

**Comment by danielmeek32** [comment]
As before I have updated the pull request.

**Comment by paramat** [comment]
Seems good, +1 if tested, needs rebase @danielmeek32

**Comment by danielmeek32** [comment]
@paramat Should be done.

**Comment by paramat** [comment]
Approved by sfan5 http://irc.minetest.net/minetest-dev/2017-08-03#i_5032465

**Comment by paramat** [comment]
271b2befaacc80f4bcc63e5cdfb31cb64ff2fec9

**Pull request closed without merging**